ssl/ssl_sess.c

```diff
@@ -163,8 +165,14 @@ SSL_SESSION *ssl_session_dup(SSL_SESSION *src, int ticket)
     if (dest->lock == NULL)
         goto err;
 
-    if (src->peer != NULL)
-        X509_up_ref(src->peer);
+    if (!CRYPTO_new_ex_data(CRYPTO_EX_INDEX_SSL_SESSION, dest, &dest->ex_data))
```

Comment:
Hmmm, no, in case `dest->lock == NULL` occurs, the SSL_SESSION_free() function would be looking at uninitialized memory during the call to CRYPTO_free_ex_data().